controllers/topolvm_node.go

```diff
@@ -0,0 +1,265 @@
+					Type: &hostPathDirectoryOrCreateType}}},
+		{Name: "lvmd-config-dir",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{Name: lvmdConfigCM}}}},
+		{Name: "lvmd-socket-dir",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{Medium: storageMedium}}},
+	}
+
+	containers := []corev1.Container{*getNodeContainer(), *getCsiRegistrarContainer(), *getLivenessProbeContainer()}
+
+	// TODO: Add the same node selector we will have in the lvmcluster CRD
+	nodeDaemonset := &v1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      TopolvmNodeDaemonsetName,
+			Namespace: lvmCluster.Namespace,
+		},
+		Spec: v1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					AppAttr: lvmCluster.Name,
+				},
+			},
+			UpdateStrategy: v1.DaemonSetUpdateStrategy{
+				Type: v1.RollingUpdateDaemonSetStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: lvmCluster.Name,
+					Labels: map[string]string{
+						AppAttr: lvmCluster.Name,
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: TopolvmNodeServiceAccount,
+					Containers:         containers,
+					Volumes:            volumes,
+					HostPID:            true,
+					Tolerations:        []corev1.Toleration{{Operator: corev1.TolerationOpExists}},
+				},
+			},
+		},
+	}
+
+	return nodeDaemonset
+}
+
+func getNodeContainer() *corev1.Container {
+	privileged := true
+	runAsUser := int64(0)
+
+	command := []string{
+		"/topolvm-node",
+		"--lvmd-socket=/run/lvmd/lvmd.sock",
+	}
+
+	requirements := corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse(TopolvmNodeCPULimit),
+			corev1.ResourceMemory: resource.MustParse(TopolvmNodeMemLimit),
+		},
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse(TopolvmNodeCPURequest),
+			corev1.ResourceMemory: resource.MustParse(TopolvmNodeMemRequest),
+		},
+	}
+
+	mountPropagationMode := corev1.MountPropagationBidirectional
+
+	volumeMounts := []corev1.VolumeMount{
+		{Name: "node-plugin-dir", MountPath: "/run/topolvm"},
+		{Name: "lvmd-socket-dir", MountPath: "/run/lvmd"},
+		{Name: "pod-volumes-dir",
+			MountPath:        fmt.Sprintf("%spods", getAbsoluteKubeletPath(CSIKubeletRootDir)),
+			MountPropagation: &mountPropagationMode},
+		{Name: "csi-plugin-dir",
+			MountPath:        fmt.Sprintf("%splugins/kubernetes.io/csi", getAbsoluteKubeletPath(CSIKubeletRootDir)),
+			MountPropagation: &mountPropagationMode},
+	}
+
+	env := []corev1.EnvVar{
+		{Name: "NODE_NAME",
+			ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "spec.nodeName"}}},
+	}
+
+	node := &corev1.Container{
+		Name:    NodeContainerName,
+		Image:   TopolvmCsiImage,
+		Command: command,
+		SecurityContext: &corev1.SecurityContext{
```

Comment:
@nbalacha : I think the SCC you requested is already is set in the node container.

Reply:
This security context is different. SCCs are described here. https://cloud.redhat.com/blog/introduction-to-security-contexts-and-sccs

Reply:
I think we can start by using the prexisting openshift sccs. One is called `privileged`

Reply:
But yeah, I think it's a different field.

Reply:
Looks like an SCC is bound to the serviceaccount via RBAC: https://developer.ibm.com/learningpaths/secure-context-constraints-openshift/scc-tutorial/#step-3-create-and-assign-an-scc

This example allows pods with the serviceAccount `scc-tutorial-sa` to use the SCC `scc-tutorial-scc`
```
 ---
 kind: Role
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: use-scc-tutorial-scc
 rules:
   - apiGroups: ["security.openshift.io"]
     resources: ["securitycontextconstraints"]
     resourceNames: ["scc-tutorial-scc"]
     verbs: ["use"]
 ---
 kind: RoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: use-scc-tutorial-scc
 subjects:
   - kind: ServiceAccount
     name: scc-tutorial-sa
 roleRef:
   kind: Role
   name: use-scc-tutorial-scc
   apiGroup: rbac.authorization.k8s.io
```

Reply:
> We need to think of how best to create these and whether these should be in the code or outside.

Outside. We have  priorities and deadlines, and that is right, but we we need to do the things as much "neutral" possible. If not what we are going to have is an operator used only by RH.

Reply:
- the ideally way is to use both SCC and PSP
- when we don't find SCC api it'll just not be deployed and so better include it in code
- not immediate though